tests/unit/db/database_tests.py

```diff
@@ -0,0 +1,480 @@
+            keys_returned = []
+            for i in rslt:
+                keys_returned.append(i['key'])
+            expected_keys = ['julia010', 'julia011', 'julia012']
+            self.assertTrue(all(x in keys_returned for x in expected_keys))
+
+    def test_keys(self):
+        """
+        Test retrieving the document keys from the database
+        """
+        self.assertEqual(self.db.keys(), [])
+        self.populate_db_with_documents(3)
+        self.assertEqual(
+            self.db.keys(remote=True),
+            ['julia000', 'julia001', 'julia002']
+            )
+
+    def test_get_non_existing_doc_via_getitem(self):
+        """
+        Test __getitem__ when retrieving a non-existing document
+        """
+        try:
+            doc = self.db['no_such_doc']
+            self.fail('Above statement should raise a KeyError')
+        except KeyError:
+            pass
+
+    def test_get_db_via_getitem(self):
+        """
+        Test __getitem__ when retrieving a document
+        """
+
+        # Add a design document
+        map_func = 'function(doc) {\n emit(doc._id, 1); \n}'
+        expected_ddoc = self.db.get_design_document('_design/ddoc01')
+        expected_ddoc.add_view('view01', map_func)
+        expected_ddoc.save()
+
+        # Add three standard documents
+        self.populate_db_with_documents(3)
+
+        # Test __get_item__ for standard document
+        doc = self.db['julia001']
+        self.assertIsInstance(doc, Document)
+        self.assertEqual(doc.get('_id'), 'julia001')
+        self.assertTrue(doc.get('_rev').startswith('1-'))
+        self.assertEqual(doc.get('name'), 'julia')
+        self.assertEqual(doc.get('age'), 1)
+
+        # Test __get_item__ for design document
+        ddoc = self.db['_design/ddoc01']
+        self.assertIsInstance(ddoc, DesignDocument)
+        self.assertTrue(ddoc, expected_ddoc)
+
+    def test_document_iteration_under_fetch_limit(self):
+        """
+        Test __iter__ works as expected when the number of documents in
+        the database is less than the database fetch limit
+        """
+        docs = []
+
+        # Check iterating when no documents exist
+        for doc in self.db:
+            self.fail('There should be no documents in the database yet!!')
+        # Check that iteration yields appropriate contents
+        self.populate_db_with_documents(3)
+        age = 0
+        for doc in self.db:
+            self.assertIsInstance(doc, Document)
+            self.assertEqual(doc['_id'], 'julia{0:03d}'.format(age))
+            self.assertTrue(doc['_rev'].startswith('1-'))
+            self.assertEqual(doc['name'], 'julia')
+            self.assertEqual(doc['age'], age)
+            docs.append(doc)
+            age += 1
+        self.assertEqual(len(docs), 3)
+        # Check that the local database object has been populated
+        # with the appropriate documents
+        expected_keys = []
+        for i in range(3):
```

Comment:
expected_keys = ['julia{0:03d}'.format(i) for i in xrange(3)] is more pythonic, though not really a big deal here... so just a suggestion